sql/analyzer/stored_procedures.go

```diff
@@ -265,34 +274,39 @@ func resolveProcedureParams(ctx *sql.Context, paramNames map[string]struct{}, pr
 
 // resolveProcedureParamsTransform resolves all of the named parameters and declared variables inside of a node.
 // In cases where an expression contains nodes, this will also walk those nodes.
-func resolveProcedureParamsTransform(ctx *sql.Context, paramNames map[string]struct{}, n sql.Node) (sql.Node, error) {
-	return plan.TransformExpressionsUp(n, func(e sql.Expression) (sql.Expression, error) {
-		switch e := e.(type) {
-		case *expression.UnresolvedColumn:
-			if strings.ToLower(e.Table()) == "" {
-				if _, ok := paramNames[strings.ToLower(e.Name())]; ok {
-					return expression.NewProcedureParam(e.Name()), nil
+func resolveProcedureParamsTransform(ctx *sql.Context, paramNames map[string]struct{}, n sql.Node) (sql.Node, bool, error) {
+	return plan.TransformUpHelper(n, func(n sql.Node) (sql.Node, bool, error) {
```

Comment:
Big red flag here

The question you should be asking is: why doesn't TransformExpressionsUpWithNode work here on its own?